power_grid_model_c/power_grid_model_cpp/include/power_grid_model_cpp/utils.hpp

```diff
@@ -0,0 +1,65 @@
+// SPDX-FileCopyrightText: Contributors to the Power Grid Model project <powergridmodel@lfenergy.org>
+//
+// SPDX-License-Identifier: MPL-2.0
+
+#pragma once
+#ifndef POWER_GRID_MODEL_CPP_UTILS_HPP
+#define POWER_GRID_MODEL_CPP_UTILS_HPP
+
+#include "basics.hpp"
+#include "handle.hpp"
+
+#include <algorithm>
+#include <complex>
+#include <limits>
+#include <type_traits>
+
+namespace power_grid_model_cpp {
+template <typename T>
+inline typename std::enable_if<std::is_floating_point<T>::value, // NOLINT(modernize-use-constraints,
+                                                                 // modernize-type-traits)
+                               bool>::type
+is_nan(T x) {
+    return std::isnan(x);
+}
+template <typename T>
+inline typename std::enable_if<std::is_floating_point<T>::value, // NOLINT(modernize-use-constraints,
+                                                                 // modernize-type-traits)
+                               bool>::type
+is_nan(std::complex<T> const& x) {
+    return is_nan(x.real()) || is_nan(x.imag());
+}
```

Comment:
```suggestion
inline bool is_nan(std::complex<double> const& x) {
    return is_nan(x.real()) || is_nan(x.imag());
}
```

Reply:
I doubt do we need `std::complex` at all in the cpp wrapper? We do not have any complex number in the C-API boundary

Reply:
We don't have any complex number in the C-API boundary indeed, but this is very useful for the validation tests, since we do get the magnitude and angle of certain attributes. In any case, I can remove and just leave the specific overload in the validation tests themselves, but this may be useful for other tests as well.